lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -29,32 +29,34 @@ def self.run(params)
         auth_token = fetch_auth_token(params[:service_credentials_file], params[:firebase_cli_token])
         fad_api_client = Client::FirebaseAppDistributionApiClient.new(auth_token, params[:debug])
 
-        # If binary is an AAB get FULL view of app which includes the aab_state
-        app_view = binary_type == :AAB ? 'FULL' : 'BASIC'
-        app = fad_api_client.get_app(app_id, app_view)
-        validate_app!(app, binary_type)
+        app = fad_api_client.get_app(app_id)
```

Comment:
Ideally we would like to no longer depend on this `v1alpha` endpoint. Consider just parsing the project number from the app ID or passing a "-" as the project number (not sure if we currently support this on the backend, but it's technically a valid use case https://google.aip.dev/159). I _think_ we could also use the `projects.androidApps` or `projects.iosApps` endpoints to get the project number.

https://firebase.google.com/docs/projects/api/reference/rest

Reply:
We need it to validate that a contact email exists.

Reply:
Yeah true but the the hope is that we could accomplish this by throwing an error on the backend instead of having to manually check the contact email address. The developer doesn't really need to know what the contact email is for the app and there is no way to update it using the API so I think this way would be preferable.

Reply:
We use the project number for IAM stuff / backend permission checking, so I'm afraid passing `-` would not work. Could we just parse out the project number from the app id, or is that considered not kosher? I think calling those other endpoints would sort of defeat the purpose, since ideally we don't have to make any extra call here.

Reply:
I parse the project number and ist atrocious: https://github.com/firebase/firebase-tools/blob/kb-use-appdistro-v1-public-api/src/appdistribution/client.ts#L71

I think we should seriously consider supporting `-` on the server (since we can extract the project number from the `appId`)... But until we do we might need to do the parsing on the client 🤮

Reply:
I have concerns with parsing the projectNumber from the appId given the format of that id isn't under our control and has changed in the past, unless we have a solid "go for it" from the providers of that ID.

Maybe we can provide an offramp for customers to provide it explicitly as a param, and fall back to parsing it from appId (with a customer-visible deprecation warning or something encouraging them to specify it explicitly), similar to how we provide a couple ways to figure out [platform](https://github.com/fastlane/fastlane-plugin-firebase_app_distribution/pull/219/files#diff-1f50a993e035d5ff62885ac3fddac664536c4ca69bb273ea666d890d3b2b5621L22)

Reply:
Another note: are we missing a public API endpoint that lets us check for the presence of a contact email that necessitates our use of the alpha endpoint?

Reply:
Nope, that’s more of a legacy concept/artifact. All onboarded apps will have a contact email set, and if they aren’t onboarded, our endpoints will return a 404 since they won’t be in spanner. I think we should just parse out the project number from the app ID, since if the app ID format ever changes, we have bigger fish to fry anyways.

Reply:
This might be worth a higher bandwidth conversation because I'm confused now :)

Reply:
Updated to parse project number from `app_id` and removed `GetApp` entirely.